core/src/main/java/org/apache/calcite/sql/type/OperandTypes.java

```diff
@@ -443,6 +443,11 @@ private boolean hasFractionalPart(BigDecimal bd) {
   public static final SqlSingleOperandTypeChecker STRING_STRING_INTEGER =
       family(SqlTypeFamily.STRING, SqlTypeFamily.STRING, SqlTypeFamily.INTEGER);
 
+  public static final SqlSingleOperandTypeChecker STRING_STRING_OPTIONAL_INTEGER =
```

Comment:
Instead of adding this in OperandTypes, we can define that an operator is optional in the function definition itself in SqlLibraryOperators.
Can we do something like this 
  OperandTypes.family(ImmutableList.of(SqlTypeFamily.INTEGER, SqlTypeFamily.STRING),
          // Second operand is optional
          number -> number == 1),

Reply:
Done